proxy/src/outbound.rs

```diff
@@ -140,7 +142,12 @@ where
 
         let loaded = tower_balance::load::WithPendingRequests::new(resolve);
 
-        let balance = tower_balance::power_of_two_choices(loaded, rand::thread_rng());
+        // we can't use `rand::thread_rng` here because the returned `Service`
+        // needs to be `Send`.
+        // XXX: StdRng::new() is very expensive. figure out another way to get
+        //      a RNG that impls `Send` but is less slow.
```

Comment:
Would love input from reviewers on this.

Reply:
Does `thread_rng()` actually do much at each call? I'd expect it to just be accessing a thread-local. In that case, we could still use it, even if this service were moving to different threads, as it'd just access a different thread local each time, no big deal. It would mean basically implementing a lazy `Rand` that calls `thread_rnd().rand_whatever()`.